Documentation/README_Interactable.md

```diff
@@ -2,416 +2,333 @@
+        lastState = state.CurrentState();
+        ...
+    }
+}
 
-    // Make the interactable have toggle capability, from code.
-    // In the gui editor it's much easier 
-    interactable.Dimensions = 2;
-    interactable.CanSelect = true;
-    interactable.CanDeselect  = true;
+public virtual void OnVoiceCommand(InteractableStates state,
+                                    Interactable source,
+                                    string command,
+                                    int index = 0,
+                                    int length = 1)
+{
+    base.OnVoiceCommand(state, source, command, index, length);
+    // voice command called, perform some action
+}  
 
-    toggleReceiver.OnSelect.AddListener(() => Debug.Log("Toggle selected"));
-    toggleReceiver.OnDeselect.AddListener(() => Debug.Log("Toggle un-selected"));
+public virtual void OnClick(InteractableStates state,
+                            Interactable source,
+                            IMixedRealityPointer pointer = null)
+{
+    base.OnClick(state, source);
+    // click called, perform some action
 }
 ```
 
-## States ##
-States are a list of terms that can be used to define interactions phases, like press or observed.
-
-<img src="../Documentation/Images/Interactable/DefaultStates.png" width="450">
+##### Displaying custom Event Receiver fields in the inspector
 
-Interactable states provide two major roles:
+*ReceiverBase* scripts use [`InspectorField`](xref:Microsoft.MixedReality.Toolkit.Utilities.Editor.InspectorField) attributes to expose custom properties in the inspector. Here's an example of Vector3 a custom property with tooltip and label information. This property will show up as configurable in the inspector when an *Interactable* GameObject is selected and has the associated *Event Receiver* type added.
 
-* Establish a list of states that are relevant for the Interactable. This list will be displayed in the themes and can also be referenced by the events.
-* Controls how different interaction phases are ranked into states. For instance, a press state is also in a focused state, but the InteractableStates class will define it is a press state based on the ranking preferences setup in the State ScriptableObject.
+```csharp
+[InspectorField(Label = "<Property label>",
+    Tooltip = "<Insert tooltip info>",
+    Type = InspectorField.FieldTypes.Vector3)]
+public Vector3 EffectOffset = Vector3.zero;
+```
 
-<img src="../Documentation/Images/Interactable/StatesScriptableObject.png" width="450">
+## How to use Interactable
 
-The InteractableStates State Model will handle any state list with a layered ranking system, starting with the most isolated state and ending with the state that could contain all other states.
+### Building a simple button
 
-The DefaultInteractableStates list contains 4 states:
+One can create a simple button by simply adding the *Interactable* component to a GameObject that has a collider on it or on a child to receive input.
```

Comment:
```suggestion
One can create a simple button by simply adding the *Interactable* component to a GameObject that has a collider on it or on a child to receive input.
```

Can you rephrase this to something like - "One can create a simple button by simply adding the *Interactable* component to a GameObject that is configured to receive input events. It can have a collider on it or on a child to receive input. If using interactable with a Unity UI based GameObjects it should be under the Canvas GameObject."

This is because i have seen a general practice of folks who prefer to use unity ui. The unity ui button does not provide as comprehensive a set of configuration options as the interactable and there is nothing really preventing Interactable from being used on unity ui components.

Reply:
Good catch, I followed existing documentation content a little too closely here